core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -154,12 +239,130 @@ oppia.directive('questionPlayer', [
+                var skillId = ctrl.questionPlayerConfig.skillList[i];
+                var description =
+                  ctrl.questionPlayerConfig.skillDescriptions[i];
+                scorePerSkillMapping[skillId] = {
+                  description: description,
+                  score: 0.0,
+                  total: 0.0
+                };
+              }
+            }
+            ctrl.scorePerSkillMapping = scorePerSkillMapping;
+          };
+
+          var calculateScores = function(questionStateData) {
+            createScorePerSkillMapping();
+            $scope.resultsLoaded = false;
+            var totalQuestions = Object.keys(questionStateData).length;
+            for (var question in questionStateData) {
+              var questionData = questionStateData[question];
+              var totalHintsPenalty = 0.0;
+              var wrongAnswerPenalty = 0.0;
+              if (questionData.answers) {
+                wrongAnswerPenalty = (
+                  (questionData.answers.length - 1) * WRONG_ANSWER_PENALTY);
+              }
+              if (questionData.usedHints) {
+                totalHintsPenalty = (
+                  questionData.usedHints.length * VIEW_HINT_PENALTY);
+              }
+              var questionScore = MAX_SCORE_PER_QUESTION;
+              if (questionData.viewedSolution) {
+                questionScore = 0.0;
+              } else {
+                // If questionScore goes negative, set it to 0
+                questionScore = Math.max(
+                  0, questionScore - totalHintsPenalty - wrongAnswerPenalty);
+              }
+              // Calculate total score
+              ctrl.totalScore += questionScore;
+
+              // Calculate scores per skill
+              if (!(questionData.linkedSkillIds)) {
+                continue;
+              }
+              for (var i = 0; i < questionData.linkedSkillIds.length; i++) {
+                var skillId = questionData.linkedSkillIds[i];
+                if (!(skillId in ctrl.scorePerSkillMapping)) {
+                  continue;
+                }
+                ctrl.scorePerSkillMapping[skillId].score += questionScore;
+                ctrl.scorePerSkillMapping[skillId].total += 1.0;
+              }
+            }
+            ctrl.totalScore = Math.round(
+              ctrl.totalScore * 100 / totalQuestions);
+            $scope.resultsLoaded = true;
+          };
+
+          var passOrFailTests = function() {
+            ctrl.passTest = true;
+            if (isInPassOrFailMode()) {
+              Object.keys(ctrl.scorePerSkillMapping).forEach(function(skillId) {
+                var correctionRate = ctrl.scorePerSkillMapping[skillId].score /
+                  ctrl.scorePerSkillMapping[skillId].total;
+                if (correctionRate < CORRECTION_RATE_TO_PASS) {
+                  ctrl.passTest = false;
+                }
+              });
+            }
+
+            if (!ctrl.passTest) {
```

Comment:
Why are you doing this?

Reply:
Sorry I meant to clear action buttons in failing mode. It was a typo. I have fixed it.